org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/launch/TestLaunchShortcut.java

```diff
@@ -0,0 +1,160 @@
+            if (selectedMethod.isPresent()) {
+                launch(ImmutableList.of(selectedMethod.get()), mode);
+            } else {
+                launch(ImmutableList.of(element), mode);
+            }
+        } else {
+            logNoTestsFound();
+        }
+    }
+
+    protected void performLaunch(List<? extends IJavaElement> elementsToLaunch, String mode) {
+        // get the right workingDirExpression for the
+        // GradleRunConfigurationAttributes
+        IJavaElement javaElement = elementsToLaunch.get(0);
+        IProject project = javaElement.getResource().getProject();
+
+        GradleRunConfigurationAttributes configurationAttributes = GradleRunConfigurationAttributes.with(
+                ImmutableList.<String> of(), project.getLocation().toOSString(), GradleDistribution.fromBuild(), null,
+                null,
+                ImmutableList.<String> of(), ImmutableList.<String> of(), true, true);
+        RunGradleJvmTestLaunchRequestJob gradleJvmTestLaunchRequestJob = new RunGradleJvmTestLaunchRequestJob(
+                elementsToLaunch, configurationAttributes);
+        gradleJvmTestLaunchRequestJob.schedule();
+    }
+
+    private void launch(List<?> elements, String mode) {
+        Builder<IJavaElement> builder = ImmutableList.builder();
+
+        for (Object selected : elements) {
+            IJavaElement element = (IJavaElement) Platform.getAdapterManager().getAdapter(selected, IJavaElement.class);
+
+            if (element != null) {
+                Optional<? extends IJavaElement> elementToLaunch = getElementToLaunch(element);
+                if (elementToLaunch.isPresent()) {
+                    builder.add(elementToLaunch.get());
+                }
+            }
+        }
+
+        ImmutableList<IJavaElement> elementsToLaunch = builder.build();
+
+        if (elementsToLaunch.isEmpty()) {
+            logNoTestsFound();
+            return;
+        }
+        performLaunch(elementsToLaunch, mode);
+    }
+
+    private Optional<IMethod> resolveSelectedMethodName(IEditorPart editor, ITypeRoot element) {
+        try {
+            ISelectionProvider selectionProvider = editor.getSite().getSelectionProvider();
+            if (selectionProvider == null) {
+                return Optional.absent();
+            }
+
+            ISelection selection = selectionProvider.getSelection();
+            if (!(selection instanceof ITextSelection)) {
+                return Optional.absent();
+            }
+
+            ITextSelection textSelection = (ITextSelection) selection;
+
+            IJavaElement elementAtOffset = SelectionConverter.getElementAtOffset(element, textSelection);
+            if (!(elementAtOffset instanceof IMethod)) {
+                return Optional.absent();
+            }
+
+            IMethod method = (IMethod) elementAtOffset;
+
+            ISourceRange nameRange = method.getNameRange();
```

Comment:
Please add comment about what it does (this this used to check if the user just selected the method name and not something in the method body)
Please add a reference where you found the same approach (just a simple comment, no javadoc)